src/core/lightning/index.ts

```diff
@@ -129,7 +132,7 @@ export const LightningMixin = (superclass) =>
 
       const { lnd, pubkey } = getActiveLnd()
 
-      const milliSats = toMilliSats(parseFloat(mtokens))
+      const milliSats = toMilliSats(parseInt(mtokens))
```

Comment:
The call to `parseInt` should be inside the `toMilliSats` function (as it can error and we want to return a validation error if it does).
If we need both conversion from `string` and `number` then we need 2 `toMilliSats` functions. `toMilliSatsFromStr` / `toMilliSatsFromNumber`.